### PR TITLE
webui: handle filenames containing backslashes properly

### PR DESCRIPTION
**Pull request by fbergkemper**
Fixes #971: Error building tree for filenames with backslashes

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

##### Tests

- [ ] Decision taken that a system- or unittest is required (if not, then remove this paragraph)
- [ ] The decision towards a systemtest is reasonable compared to a unittest
- [ ] Testname matches exactly what is being tested
- [ ] Output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: webui/module/Restore/src/Restore/Controller/RestoreController.php [hunk 1]
```diff
@@ -588,7 +588,7 @@ private function buildSubtree()
           --$dnum;
           $items .= '{';
           $items .= '"id":"-' . $dir['pathid'] . '"';
-          $items .= ',"text":"' . preg_replace('/[\x00-\x1F\x7F]/', '', str_replace('"', '\"', $dir["name"])) . '"';
+          $items .= ',"text":"' . preg_replace('/[\x00-\x1F\x7F]/', '', str_replace('"', '\"', str_replace('\\', '\\\\', $dir["name"]))) . '"';
```
Comment (arogge) [inline webui/module/Restore/src/Restore/Controller/RestoreController.php:591]: that's the same three substitutions here and on line 611.
Don't you think this deserves its own function?

I'd also merge the two calls to `str_replace()` with a single call to `strtr()` with the second signature (i.e. that one with the `array $replace_pairs`.

Comment (fbergkemper) [reply webui/module/Restore/src/Restore/Controller/RestoreController.php]: Good suggestion.

Comment (fbergkemper) [reply webui/module/Restore/src/Restore/Controller/RestoreController.php]: Adapted.

**Review by arogge — APPROVED** [review]
Looks good!

**Pull request merged**